### PR TITLE
Extension storage path to product image #30

### PR DESCRIPTION
**Pull request by n2pro**

### CODE REVIEW

File: VirtoCommerce.CatalogModule.Web/Scripts/resources/categories.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 ﻿angular.module('virtoCommerce.catalogModule')
 .factory('virtoCommerce.catalogModule.categories', ['$resource', function ($resource) {
 
-	return $resource('api/catalog/categories/:id', { categoryId: '@Id' }, {
+    return $resource('api/catalog/categories/:id', { categoryId: '@Id' }, {
```
Comment (tatarincev) [inline VirtoCommerce.CatalogModule.Web/Scripts/resources/categories.js]: This method declaration is redundant. 
Follow construction will work without this definition `categories.get({ id: blade.currentEntityId }`

File: VirtoCommerce.CatalogModule.Web/Scripts/blades/images-add.js [hunk 2]
```diff
@@ -2,8 +2,8 @@
 .controller('virtoCommerce.catalogModule.imagesAddController',
     ['$scope', '$filter', '$translate', 'FileUploader', 'platformWebApp.dialogService',
         'platformWebApp.bladeNavigationService', 'platformWebApp.authService',
-        'platformWebApp.assets.api', 'virtoCommerce.catalogModule.imageTools', 'platformWebApp.settings',
-        function ($scope, $filter, $translate, FileUploader, dialogService, bladeNavigationService, authService, assets, imageTools, settings) {
+        'platformWebApp.assets.api', 'virtoCommerce.catalogModule.imageTools', 'platformWebApp.settings', 'virtoCommerce.catalogModule.categories',
+        function ($scope, $filter, $translate, FileUploader, dialogService, bladeNavigationService, authService, assets, imageTools, settings, categories) {
```
Comment (tatarincev) [inline VirtoCommerce.CatalogModule.Web/Scripts/blades/images-add.js]: Redundant `categories` injection is not used anywhere

File: VirtoCommerce.CatalogModule.Web/Scripts/blades/images-add.js [hunk 3]
```diff
@@ -39,7 +40,7 @@
                         removeAfterUpload: true
                     });
 
-                    uploader.url = getImageUrl(item.code, blade.imageType).relative;
+                    uploader.url = getImageUrl(item.code, blade.imageType, item.category).relative;
```
Comment (tatarincev) [inline VirtoCommerce.CatalogModule.Web/Scripts/blades/images-add.js]: This controller reused in other places. and `item` here might be a category type.
I this is a not correct way to try get the storage path from element here, the better solution would be to pass a storage folder path as a special parameter to this controller.

File: VirtoCommerce.CatalogModule.Web/Scripts/resources/categories.js (hunk 1 above)
Comment (n2pro) [reply VirtoCommerce.CatalogModule.Web/Scripts/resources/categories.js]: remove this get

File: VirtoCommerce.CatalogModule.Web/Scripts/blades/category-detail.js [hunk 4]
```diff
@@ -34,13 +35,18 @@
         blade.title = data.name;
         blade.isLoading = false;
         blade.securityScopes = data.securityScopes;
+        blade.currentEntity.folderPath = (data.catalogId ? (getShortCatalogId(data.catalogId) + '/') : '') + data.code;
```
Comment (tatarincev) [inline VirtoCommerce.CatalogModule.Web/Scripts/blades/category-detail.js]: This line leads to forcing category modification it is really the bug. The folder path initialization code should be moved to the CategoryImageWidget controller.

File: VirtoCommerce.CatalogModule.Web/Scripts/blades/item-detail.js [hunk 5]
```diff
@@ -30,6 +29,10 @@
             blade.currentEntity = blade.item;
             blade.origItem = data;
             blade.isLoading = false;
+
+            //set folderPath
```
Comment (tatarincev) [inline VirtoCommerce.CatalogModule.Web/Scripts/blades/item-detail.js]: This line leads to forcing Item modification it is really the bug. The folder path initialization code should be moved to the ItemImageWidget controller.

**Pull request merged**